profiles/killbill/src/main/java/org/killbill/billing/server/notifications/PushNotificationKey.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2014-2016 Groupon, Inc
+ * Copyright 2014-2016 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.server.notifications;
+
+import java.util.UUID;
+
+import org.killbill.notificationq.api.NotificationEvent;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class PushNotificationKey implements NotificationEvent {
+
+    private final UUID tenantId;
+    private final UUID accountId;
+    private final String eventType;
+    private final String objectType;
+    private final UUID objectId;
+    private Integer attemptNumber;
+    private final String url;
```

Comment:
Nit: I think it would be nicer to keep the field as private final and provide a constructor that takes both a `PushNotificationKey` and the next `attemptNumber `:

```
public PushNotificationKey(final PushNotificationKey input, final Integer nextAttemptNumber) {
 ....
}
```